contracts/ControllerCurveStrat.sol

```diff
@@ -9,43 +9,99 @@ contract ControllerCurveStrat is ControllerStratAbs {
     using SafeERC20 for IERC20;
     using SafeERC20 for IERC20Metadata;
 
-    // Test
-    address constant public WNATIVE = address(0x73511669fd4dE447feD18BB79bAFeAC93aB7F31f);
-    address constant public CRV = address(0xED8CAB8a931A4C0489ad3E3FB5BdEA84f74fD23E);
-    address constant public ETH = address(0x73511669fd4dE447feD18BB79bAFeAC93aB7F31f); // same than wNative
-    address constant public BTCCRV = address(0x40bde52e6B80Ae11F34C58c14E1E7fE1f9c834C4); // same than CurvePool
-    address constant public CURVE_POOL = address(0x40bde52e6B80Ae11F34C58c14E1E7fE1f9c834C4);
-    address constant public GAUGE = address(0xE9061F92bA9A3D9ef3f4eb8456ac9E552B3Ff5C8);
-    address constant public GAUGE_FACTORY = address(0xA7c8B0D74b68EF10511F27e97c379FB1651e1eD2);
-
-    constructor(address _controller, address _exchange, address _treasury)
-        ControllerStratAbs(
-            IERC20Metadata(0x6d925938Edb8A16B3035A4cF34FAA090f490202a), // BTC
-            _controller,
-            _exchange,
-            _treasury
-        ) {}
-
-    function identifier() external pure returns (string memory) {
-        return string("Ren@Curve#1.0.0");
+    address public crvToken;
+    address public pool;
+    address public swapPool;
+    address public gauge;
+    address public gaugeFactory;
+
+    uint private immutable poolSize;
+    int128 private immutable tokenIndex; // want token index in the pool
+
+    // gaugeType { STAKING = 0, CHILD_STAKING = 1 }
+    uint private immutable gaugeType; // 0 == staking, 1 == child
+
+    constructor(
+        IERC20Metadata _want,
+        address _controller,
+        address _exchange,
+        address _treasury,
+        address _crvToken,
+        address _pool,
+        address _swapPool,
+        address _gauge,
+        address _gaugeFactory,
+        uint _gaugeType,
+        uint _poolSize,
+        int128 _tokenIndex
+    ) ControllerStratAbs(_want, _controller, _exchange, _treasury) {
+        require(_crvToken != address(0), "crvToken !ZeroAddress");
+        require(_pool != address(0), "pool !ZeroAddress");
+        require(_swapPool != address(0), "swapPool !ZeroAddress");
+        require(_gauge != address(0), "gauge !ZeroAddress");
+        require(_gaugeFactory != address(0), "gaugeFactory !ZeroAddress");
+        require(_gaugeType < 2, "gaugeType unknown");
+        require(_poolSize > 0, "poolSize is zero");
+        require(uint(int256(_tokenIndex)) < _poolSize, "tokenIndex out of bounds");
+
```

Comment:
I'ld add a few checks with the interface here to be sure that gauge is a gauge and factory is a factory. Just to be sure that it doesn't raise =)
`ICurveGauge(_gauge).claimable_tokens(address(this))`

And can we get tokenIndex & poolSize from pool?